dokka-subprojects/analysis-kotlin-descriptors-compiler/src/main/kotlin/org/jetbrains/dokka/analysis/kotlin/descriptors/compiler/impl/DescriptorSampleAnalysisEnvironment.kt

```diff
@@ -180,16 +193,90 @@ internal class DescriptorSampleAnalysisEnvironment(
             .trimIndent()
     }
 
-    private fun getSampleBody(sampleElement: PsiElement): String {
-        return when (sampleElement) {
+    private fun getSampleBody(psiElement: PsiElement): String {
+        return when (psiElement) {
             is KtDeclarationWithBody -> {
-                when (val bodyExpression = sampleElement.bodyExpression) {
-                    is KtBlockExpression -> bodyExpression.text.removeSurrounding("{", "}")
-                    else -> bodyExpression!!.text
+                val bodyExpression = psiElement.bodyExpression
+                val bodyExpressionText = bodyExpression!!.buildSampleText()
+                when (bodyExpression) {
+                    is KtBlockExpression -> bodyExpressionText.removeSurrounding("{", "}")
+                    else -> bodyExpressionText
                 }
             }
 
-            else -> sampleElement.text
+            else -> psiElement.text
         }
     }
+
+    private fun PsiElement.buildSampleText(): String {
+        val sampleBuilder = SampleBuilder(sampleRewriter)
+        this.accept(sampleBuilder)
+
+        sampleBuilder.errors.forEach {
+            val sw = StringWriter()
+            val pw = PrintWriter(sw)
+            it.e.printStackTrace(pw)
+
+            dokkaLogger.warn("${containingFile.name}: (${it.loc}): Exception thrown while converting \n```\n${it.text}\n```\n$sw")
+        }
+        return sampleBuilder.text
+    }
 }
+
+private class SampleBuilder(private val sampleRewriter: SampleRewriter?) : KtTreeVisitorVoid() {
+    val builder = StringBuilder()
+    val text: String
+        get() = builder.toString()
+
+    val errors = mutableListOf<ConvertError>()
+
+    data class ConvertError(val e: Exception, val text: String, val loc: String)
+
+    override fun visitCallExpression(expression: KtCallExpression) {
+        val callRewriter = sampleRewriter?.functionCallRewriters?.get(expression.calleeExpression?.text)
+        if(callRewriter != null) {
+            val rewritedResult = callRewriter.rewrite(
+                argumentList = expression.valueArguments.map { it.text ?: "" },
```

Comment:
I think, for example, in case of some incorrect (can not be compiled) sample snippet `fun f(,,,,)` but I have not checked this case.

Reply:
I had a similar question, as I thought it would be weird to see just an empty string in `argumentList` as it's not mentioned in the KDoc

With the case Vadim provided, not sure what's better though: to pass an empty string or to exclude it at all (via `mapNotNull`), as both have disadvantages

Reply:
Sorry for the misleading.  I have checked :`fun f(,,,,)` just returns an empty string for the parameter. I will put this case into the KDoc.

An answer is [here](https://github.com/JetBrains/intellij-community/blob/a6084634718b8609a3b91758d2125665a6bdc3b6/platform/core-impl/src/com/intellij/extapi/psi/ASTDelegatePsiElement.java#L133), it returns  `getNode().getText()` that is marked with `NotNull`.
So there is no real case for null.